pydantic/functional_serializers.py

```diff
@@ -282,35 +279,67 @@ def dec(
             when_used=when_used,
             check_fields=check_fields,
         )
-        return _decorators.PydanticDescriptorProxy(f, dec_info)
+        return _decorators.PydanticDescriptorProxy(f, dec_info)  # pyright: ignore[reportArgumentType]
 
-    return dec
+    return dec  # pyright: ignore[reportReturnType]
 
 
-FuncType = TypeVar('FuncType', bound=Callable[..., Any])
+if TYPE_CHECKING:
+    # The first argument in the following callables represent the `self` type:
+
+    ModelPlainSerializer: TypeAlias = Callable[[Any, SerializationInfo], Any]
+    """A model serializer method with the `info` argument, in `plain` mode."""
+
+    ModelPlainSerializerWithoutInfo: TypeAlias = Callable[[Any], Any]
+    """A model serializer method without the `info` argument, in `plain` mode."""
+
+    AnyModelPlainSerializer: TypeAlias = 'ModelPlainSerializer | ModelPlainSerializerWithoutInfo'
+    """A model serializer method in `plain` mode."""
```

Comment:
Re my comment above discussing the use of `Any`, what about:

`ModelPlainSerializerWithInfo`, `ModelPlainSerializerWithoutInfo`, `ModelPlainSerializer`? And a similar pattern above?

Reply:
Yep `ModelPlainSerializer` looks good, I'll use this pattern for the other changes as well.